lib/ts/recipe/emailverification/api/emailVerify.ts

```diff
@@ -42,10 +45,18 @@ export default async function emailVerify(apiImplementation: APIInterface, optio
             });
         }
 
+        const session = await Session.getSession(
+            options.req,
+            options.res,
+            { overrideGlobalClaimValidators: () => [] },
+            userContext
+        );
```

Comment:
Since the `verifyEmailPOST` API takes an optional session, you should pass the `requiredAuth: false` in here